### DISCUSSION TITLE
empty click listener on root __next element

### DISCUSSION THREAD

**Discussion by mrr11k** [vercel/next.js#52884] [Help]
### Summary

Hi everyone,
I'm experiencing an issue, that on some pages I'm seeing an empty click listener attached to the root div element with id="__next". The handler is located in the framework-xyz.js file.
Is there a way to debug properly why and where this handler is coming from?
I tried to use `next build --no-mangling`, to get an understandable build output. But i doesn't seem to have an effect. 

Any tips appreciated. Thanks in advance 🙂
![image](https://github.com/vercel/next.js/assets/20047836/5f418100-f490-40b3-809a-a8a29d65ef4b)
![image](https://github.com/vercel/next.js/assets/20047836/505d8928-bba1-48a3-bba4-a86f555f3f48)

**Comment by icyJoseph** [comment]
I do believe this is part of event delegation. Since React 17, React attaches event listeners to the root of the app, and not the document element. 

- https://blog.saeloun.com/2021/07/08/react-17-event-delagation/#:~:text=Event%20Delegation%20in%20React%2017,('root')%3B%20ReactDOM.

My guess is that the pages that show this behavior are using `onClick` handlers? 

You can select the root (#__next) on the browser devTools Elements, and then on the console run `getEventListeners($0)`, that'll show you a whole bunch of listeners attached to the root of the app.

**Comment by mrr11k** [comment]
Hi,
thanks for the help. In the end we found out, it was caused by a hydration error.
